packages/manager/src/features/Account/SwitchAccounts/useParentTokenManagement.tsx

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+
+import { switchAccountSessionContext } from 'src/context/switchAccountSessionContext';
+import { isParentTokenValid } from 'src/features/Account/utils';
+
+export const useParentTokenManagement = ({
+  isProxyUser,
+}: {
+  isProxyUser: boolean;
+}) => {
+  const [isParentTokenExpired, setIsParentTokenExpired] = React.useState(false);
+  const sessionContext = React.useContext(switchAccountSessionContext);
+
+  const checkParentToken = async () => {
+    const isExpired = !isParentTokenValid();
+    setIsParentTokenExpired(isExpired);
+  };
+
+  React.useEffect(() => {
+    if (isProxyUser) {
+      checkParentToken();
+    }
```

Comment:
Can we inline this? The function name `checkParentToken` is a bit misleading since it's also setting a value

```suggestion
    if (isProxyUser) {
      const isExpired = !isParentTokenValid();
      setIsParentTokenExpired(isExpired);
    }
```